packages/ui-components/src/components/time-picker/time-picker.tsx

```diff
@@ -290,9 +288,9 @@ export class KvTimePicker implements ITimePicker, ITimePickerEvents {
 	// Components config methods
 	private isApplyButtonDisabled() {
```

Comment:
This method could use some refactor, don't you agree?
Maybe something of this sort:

```ts
        private isApplyButtonDisabled() {
		if (!this.selectedTimeState) {
			return true;
		}

		if (!this.selectedTimeState) {
			return false;
		}

		if (this.selectedTimeState.range) {
			const { range: currentRange } = this.selectedTimeState;
			const { range: newRange } = this.selectedTimeOption;

			if (currentRange.length === FULL_RANGE_SIZE && hasRangeChanged(currentRange, newRange) && validateNewRange(newRange)) {
				return false;
			}
		}

		if (this.selectedTimeState.timezone?.name !== this.selectedTimeOption.timezone?.name) {
			return false;
		}

		return true;
	}
```